qubes/vm/qubesvm.py

```diff
@@ -839,24 +804,38 @@ def start(self, start_guid=True, notify_function=None,
                 pre_event=True,
                 start_guid=start_guid, mem_required=mem_required)
 
-            yield from self.storage.verify()
-
-            if self.netvm is not None:
-                # pylint: disable = no-member
-                if self.netvm.qid != 0:
-                    if not self.netvm.is_running():
-                        yield from self.netvm.start(start_guid=start_guid,
-                            notify_function=notify_function)
-
-            qmemman_client = yield from asyncio.get_event_loop().\
-                run_in_executor(None, self.request_memory, mem_required)
+            try:
+                yield from self.storage.verify()
+
+                if self.netvm is not None:
+                    # pylint: disable = no-member
+                    if self.netvm.qid != 0:
+                        if not self.netvm.is_running():
+                            yield from self.netvm.start(start_guid=start_guid,
+                                notify_function=notify_function)
+
+                qmemman_client = yield from asyncio.get_event_loop().\
+                    run_in_executor(None, self.request_memory, mem_required)
+
+            except Exception as exc:
+                # let anyone receiving domain-pre-start know that startup failed
+                yield from self.fire_event_async('domain-start-failed',
+                    reason=str(exc))
```

Comment:
Why is this `reason=str(exc)` and not just `reason=exc`? Someone may extract something useful from the exception object (like `errno` or whatever).

Reply:
This event is mostly for those listening on `admin.Events`. I want to be sure it get serialized using sane way. For those wanting actual exception object (so, running inside qubesd process), there is next line: `raise`.